specs/agents/field-limits.md

```diff
@@ -0,0 +1,31 @@
+## Field limits
+
+The maximum length of metadata, transaction, span, et al fields are determined
+by the [APM Server Events Intake API schema](https://www.elastic.co/guide/en/apm/server/current/events-api.html).
+Fields that are names or identifiers of some resource of typically limited to
+1024 unicode characters.
+
+### `long_field_max_length` configuration
+
+Some APM event fields are not limited in the APM server intake API schema.
+Agents SHOULD limit the maximum length of the following fields by truncating.
+
+- `{transaction,error}.context.request.body`
+- `{transaction,span,error}.context.message.body`
+- `span.context.db.statement`
+- `error.exception.message`
+- `error.log.message`
+
+Agents MAY support the `long_field_max_length` configuration option to allow
+the user to configure this maximum length. This option defines a maximum number
+of unicode characters for each field.
+
+|                |   |
+|----------------|---|
+| Type           | `Integer` |
+| Default        | `10000` |
+| Dynamic        | `true` |
+| Central config | `true` |
```

Comment:
Regarding including it in central config:
- if `true` it might be confusing for users, especially if it's not supported with all agents, also, we would have to explain them what does it do exactly and when to change the default value, and thus brings a nice opportunity for mis-configuration
- if `false` : we can still add it later when (and if) we have broad agent support and it proves to be relevant.

Thus the question is that does the initial requirement (for example a customer request) would have been impacted by the ability to configure this directly from Kibana ? If it's just a low-level detail that users rarely have to change themselves I would instead not include it in Kibana.

Also, I start to feel that the question of UI complexity on Kibana could be different if we had the possibility to define "advanced" parameters like this that are not visible by default. Also, having the ability to define per-platform usage of configuration could also help, or even having an almost plain-text input for "advanced configuration for advanced users" that is very rarely used and that is passed almost as-is to agents.

Reply:
I'm sold on having this `false` to start. Done in commit 2deecbe304b0d2592c998ab8a8a25f2e988cfd3a.